Failure when SCC is enabled with -XX:StartOpenJ9ExperimentalFlightRecording

```
ava.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:575)
	at java.base/java.lang.JFRHelpers.initJFRCmdlineOptions(JFRHelpers.java:129)
	at java.base/java.lang.JFRHelpers.initJFRv2(JFRHelpers.java:229)
Caused by: java.lang.InternalError
	at java.base/java.lang.VMAccess.findClassOrNull(Native Method)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:1099)
	at jdk.jfr/jdk.jfr.internal.instrument.JDKEvents.<clinit>(JDKEvents.java:69)
	at jdk.jfr/jdk.jfr.internal.PlatformRecorder.<init>(PlatformRecorder.java:86)
	at jdk.jfr/jdk.jfr.FlightRecorder.getFlightRecorder(FlightRecorder.java:179)
	at jdk.jfr/jdk.jfr.internal.dcmd.DCmdStart.initializeWithForcedInstrumentation(DCmdStart.java:284)
	at jdk.jfr/jdk.jfr.internal.dcmd.DCmdStart.execute(DCmdStart.java:132)
	at jdk.jfr/jdk.jfr.internal.dcmd.AbstractDCmd.execute(AbstractDCmd.java:88)
	... 6 more
Exception in thread "main" java.lang.InternalError: java.lang.InternalError
	at java.base/java.lang.JFRHelpers.initJFRCmdlineOptions(JFRHelpers.java:138)
	at java.base/java.lang.JFRHelpers.initJFRv2(JFRHelpers.java:229)
Caused by: java.lang.InternalError
	at java.base/java.lang.VMAccess.findClassOrNull(Native Method)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:1099)
```

```
xmm15=000001420000013b (f: 315.000000, d: 6.832826e-312)
Module=/home/tobi/openj9-openjdk-jdk17/build/linux-x86_64-server-release/images/jdk/lib/default/libj9vm29.so
Module_base_address=00007D7F8BC00000
Target=2_90_20260408_000000 (Linux 6.8.0-107-generic)
CPU=amd64 (16 logical CPUs) (0xfb2d1b000 RAM)
----------- Stack Backtrace -----------
_ZN32VM_BytecodeInterpreterCompressed3runEP10J9VMThread+0x69c3 (0x00007D7F8BCC2903 [libj9vm29.so+0xc2903])
bytecodeLoopCompressed+0xca (0x00007D7F8BCBBEAA [libj9vm29.so+0xbbeaa])
 (0x00007D7F8BD68402 [libj9vm29.so+0x168402])
---------------------------------------
JVMDUMP039I Processing dump event "gpf", detail "" at 2026/04/13 10:16:03 - please wait.
JVMDUMP056I Processed dump event "gpf", detail "" at 2026/04/13 10:16:03 (0.002 seconds).
```

To reproduce run with `java -Djfr.unsupported.vm=false -XX:StartOpenJ9ExperimentalFlightRecording  -version`

--- a/runtime/bcutil/defineclass.c
+++ b/runtime/bcutil/defineclass.c
@@ -160,10 +160,23 @@ internalDefineClass(
 		} else if (NULL != superClass) {
 			J9Class *internalEventClass = J9VMJAVALANGCLASS_VMREF(vmThread, J9_JNI_UNWRAP_REFERENCE(vm->jfrState.jfrInternalEventClassRef));
 			if (isSameOrSuperClassOf(internalEventClass, superClass)) {
-				U_8* jfrModifiedBytes = NULL;
+				U_8 *jfrModifiedBytes = NULL;
 				UDATA jfrModifiedBytesLength = 0;
+				U_8 *upcallClassBytes = classData;
+				U_32 upcallClassBytesLength = classDataLength;
+
+				if (J9_ARE_ALL_BITS_SET(options, J9_FINDCLASS_FLAG_SHRC_ROMCLASS_EXISTS)) {
+					PORT_ACCESS_FROM_JAVAVM(vm);
+					UDATA rc = BCT_ERR_NO_ERROR;
+					rc = j9bcutil_transformROMClass(vm, PORTLIB, (J9ROMClass *)classData, &upcallClassBytes, &upcallClassBytesLength);
+					if (BCT_ERR_NO_ERROR != result) {
+						Trc_BCU_internalLoadROMClass_ErrorInRecreatingClassfile(vmThread, classData, rc);
+						vmFuncs->setCurrentExceptionUTF(vmThread, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, NULL);
+						return NULL;
+					}
+				}
 				omrthread_monitor_exit(vm->classTableMutex);
-				vmFuncs->jvmUpcallsEagerByteInstrumentation(vmThread, superClass, className, (U_16)classNameLength, classLoader, classData, classDataLength, &jfrModifiedBytes, &jfrModifiedBytesLength);
+				vmFuncs->jvmUpcallsEagerByteInstrumentation(vmThread, superClass, className, (U_16)classNameLength, classLoader, upcallClassBytes, upcallClassBytesLength, &jfrModifiedBytes, &jfrModifiedBytesLength);
 				omrthread_monitor_enter(vm->classTableMutex);
 				if ((NULL == jfrModifiedBytes) || (0 == jfrModifiedBytesLength)) {
 					omrthread_monitor_exit(vm->classTableMutex);

--- a/test/functional/cmdLineTests/jfr/jfrV2.xml
+++ b/test/functional/cmdLineTests/jfr/jfrV2.xml
@@ -26,20 +26,20 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-ex
 
 <suite id="JFR v2 Tests" timeout="3000">
 	<test id="Test JFRv2 cmdline option - no options">
-		<command>$EXE$ -XX:+EnableOpenJ9ExperimentalFlightRecording -XX:StartFlightRecording -Xshare:off -Djfr.unsupported.vm=false  -version</command>
+		<command>$EXE$ -XX:+EnableOpenJ9ExperimentalFlightRecording -XX:StartFlightRecording -Djfr.unsupported.vm=false  -version</command>
 		<output type="success" caseSensitive="yes" regex="no">Eclipse OpenJ9 VM</output>
 		<output type="required" caseSensitive="yes" regex="no">OpenJDK Runtime Environment</output>
 	</test>
 	<!-- See https://github.com/eclipse-openj9/openj9/issues/23980
 	<test id="Test JFRv2 cmdline option - memcheck">
-		<command>$EXE$ -XX:+EnableOpenJ9ExperimentalFlightRecording -Xcheck:memory -Dibm.java9.forceCommonCleanerShutdown=true -Xint -XX:StartFlightRecording -Xshare:off -Djfr.unsupported.vm=false  -version</command>
+		<command>$EXE$ -XX:+EnableOpenJ9ExperimentalFlightRecording -Xcheck:memory -Dibm.java9.forceCommonCleanerShutdown=true -Xint -XX:StartFlightRecording -Djfr.unsupported.vm=false  -version</command>
 		<output type="success" caseSensitive="yes" regex="no">Eclipse OpenJ9 VM</output>
 		<output type="required" caseSensitive="yes" regex="no">OMR</output>
 		<output type="failure" caseSensitive="yes" regex="no">bytes were not freed before shutdown</output>
 	</test>
 	-->
 	<test id="Test JFRv2 with file">
-		<command>$EXE$ -XX:+EnableOpenJ9ExperimentalFlightRecording -XX:StartFlightRecording=filename=experimentalRecording.jfr -Xshare:off -Djfr.unsupported.vm=false -cp $RESJAR$ org.openj9.test.WorkLoad</command>
+		<command>$EXE$ -XX:+EnableOpenJ9ExperimentalFlightRecording -XX:StartFlightRecording=filename=experimentalRecording.jfr -Djfr.unsupported.vm=false -cp $RESJAR$ org.openj9.test.WorkLoad</command>
 		<output type="success" caseSensitive="yes" regex="no">All runs complete</output>
 		<output type="required" caseSensitive="yes" regex="no">0 / 100 complete</output>
 	</test>
@@ -60,13 +60,13 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-ex
 		<output type="required" caseSensitive="yes" regex="no">JVMInformation</output>
 	</test>
 	<test id="Test JFRv2 with JMX">
-		<command>$EXE$ -Dcom.sun.management.jmxremote -Djava.rmi.server.hostname=127.0.0.1 -Dcom.sun.management.jmxremote.local.only=false  -Dcom.sun.management.jmxremote.port=9898 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -XX:+EnableOpenJ9ExperimentalFlightRecording -XX:StartFlightRecording=filename=test1.jfr -Xshare:off -Djfr.unsupported.vm=false  -version</command>
+		<command>$EXE$ -Dcom.sun.management.jmxremote -Djava.rmi.server.hostname=127.0.0.1 -Dcom.sun.management.jmxremote.local.only=false  -Dcom.sun.management.jmxremote.port=9898 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -XX:+EnableOpenJ9ExperimentalFlightRecording -XX:StartFlightRecording=filename=test1.jfr -Djfr.unsupported.vm=false  -version</command>
 		<output type="success" caseSensitive="yes" regex="no">Eclipse OpenJ9 VM</output>
 		<output type="required" caseSensitive="yes" regex="no">OMR</output>
 	</test>
 	<!-- See https://github.com/eclipse-openj9/openj9/issues/23980
 	<test id="Test JFRv2 with JMX and memcheck">
-		<command>$EXE$ -Xcheck:memory -Dibm.java9.forceCommonCleanerShutdown=true -Xint -Dcom.sun.management.jmxremote -Djava.rmi.server.hostname=127.0.0.1 -Dcom.sun.management.jmxremote.local.only=false  -Dcom.sun.management.jmxremote.port=9898 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -XX:+EnableOpenJ9ExperimentalFlightRecording -XX:StartFlightRecording=filename=test1.jfr -Xshare:off -Djfr.unsupported.vm=false  -version</command>
+		<command>$EXE$ -Xcheck:memory -Dibm.java9.forceCommonCleanerShutdown=true -Xint -Dcom.sun.management.jmxremote -Djava.rmi.server.hostname=127.0.0.1 -Dcom.sun.management.jmxremote.local.only=false  -Dcom.sun.management.jmxremote.port=9898 -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -XX:+EnableOpenJ9ExperimentalFlightRecording -XX:StartFlightRecording=filename=test1.jfr -Djfr.unsupported.vm=false  -version</command>
 		<output type="success" caseSensitive="yes" regex="no">Eclipse OpenJ9 VM</output>
 		<output type="required" caseSensitive="yes" regex="no">OMR</output>
 		<output type="failure" caseSensitive="yes" regex="no">bytes were not freed before shutdown</output>

